docs/topic-guides/training-units.txt

```diff
@@ -0,0 +1,185 @@
+
+When training a single model or performing a hyperparameter search, we
+need to specify how much data a model should be trained on before
+certain actions are taken (e.g., before training is terminated or before
+a checkpoint of the state of the model is taken). Determined supports a
+flexible system of *training units* to specify this. Training units can
+be specified as *records*, *batches*, or *epochs*:
+
+-  ``records``: A *record* is a single labeled example (sometimes called
+   a sample).
+
+-  ``batches``: A *batch* is a group of records. The number of records
+   in a batch is configured via the ``global_batch_size``
+   hyperparameter.
+
+-  ``epochs``: An *epoch* is a single copy of the entire training data
+   set. The number of records in an epoch is configured via the
+   ``records_per_epoch`` experiment configuration setting.
+
+Training units must always be positive integers.
+
+Several :ref:`experiment configuration parameters
+<experiment-configuration>` can be specified in terms of training units,
+including:
+
+-  :ref:`min_validation_period
+   <experiment-config-min-validation-period>`
+
+-  :ref:`min_checkpoint_period
+   <experiment-config-min-checkpoint-period>`
+
+-  :ref:`searcher.max_length
+   <experiment-configuration_single-searcher-max-length>` in most
+   searchers
+
+-  :ref:`searcher.budget
+   <experiment-configuration_adaptive-advanced-budget>` when using the
+   ``adaptive`` searcher
+
+-  :ref:`searcher.length_per_round
+   <experiment-configuration_length-per-round>` when using the ``pbt``
+   searcher
+
+For example, an experiment that trains a single trial on 10,000 labeled
+examples can be configured as follows:
+
+.. code:: yaml
+
+   searcher:
+     name: single
+     metric: validation_error
+     max_length:
+       records: 10000
+     smaller_is_better: true
+
+More examples and details on each of the types of training units can be
+seen below.
+
+This feature is designed to allow users to configure their experiments
+using whatever unit is most familiar for the task at hand. In most
+cases, a value expressed using one type of training unit can be
+converted to a different type of training unit with identical behavior,
+with a few caveats:
+
+-  Because training units must be positive integers, converting between
+   quantities of different types is not always possible. For example,
+   converting 50 ``records`` into batches is not possible if the batch
+   size is 64.
+
+-  When doing a hyperparameter search over a range of values for
+   ``global_batch_size``, values specified in ``batches`` will differ
+   between trials of the search, and hence cannot be converted to a
+   fixed number of records or epochs.
+
+-  When using adaptive searchers, such as :ref:`adaptive_asha
+   <experiment-configuration-searcher-adaptive>`, a single training unit
+   is treated as atomic (unable to be divided into fractional parts)
+   when dividing the ``max_length`` into rungs. This may result in
```

Comment:
suggest explaining "rungs" here since i'd expect many users won't know what it means. something like:

... when dividing ``max_length`` into the series of rounds (or _rungs_) by which we early-stop underperforming trials.  This rounding up per rung may result in unexpected behavior when configuring ``max_length`` in terms of a small number of large epochs or batches.